draft-ietf-privacypass-architecture.md

```diff
@@ -544,6 +548,88 @@ and guidelines on how to deploy the protocol to minimize any privacy impacts.
 Any extension to the Privacy Pass protocol MUST adhere to the guidelines
 specified in {{issuer-role}} for managing Issuer public key data.
 
+## Information Flow {#flow}
+
+The end-to-end run of redemption and issuance protocol involves information
+flowing between Issuer, Origin, Client, and Attester. That information can
+have implications on the privacy goals that Privacy Pass aims to provide
+as outlined in {{privacy-and-trust}}. In this section, we describe the flow
+of information between each party. How this information affects the privacy
+goals in particular deployment models is further discussed in {{deployment}}.
+
+### Token Challenge Flow {#challenge-flow}
+
+To use Privacy Pass, Origins choose an Issuer from which they are willing to
+accept tokens. Origins then construct a token challenge using this specified
+Issuer and the redemption context it shares with the Client. This token
+challenge is then delivered to a Client. The token challenge conveys
+information about the Issuer and the redemption context, such as whether the
+Origin desires a per-Origin or cross-Origin token. Any entity that sees
+the token challenge might learn things about the Client as known to the Origin.
+This is why input secrecy is a requirement for issuance protocols, as it
+ensures that the challenge is not directly available to the Issuer.
+
+### Attestation and Issuance Flow {#issue-flow}
+
+Clients use the issuance protocol to produce a token bound to a token
+challenge. In doing so, there are several ways in which the issuance protocol
+contributes information to the attestation context and issuance context.
```

Comment:
I'm still not convinced by this being a single section.  The attestation context only gains information from the issuance context transitively.  That is, if you need to use attestation method W to execute issuance protocol X and you need issuance protocol X to get token Y to answer challenge Z, then you can trace the way from Z to W, but each step is its own and the client is more or less in control of whether to proceed in each, provided that it understands what information carries over.

Reply:
The attestation context does sometimes gain information explicitly from issuance, as in the rate limited protocol. I'll try to clarify that in the definitions.